proxy/api/src/http.rs

```diff
@@ -114,6 +123,22 @@ fn with_owner_guard(ctx: context::Context) -> BoxedFilter<(coco::user::User,)> {
         .boxed()
 }
 
+/// Asserts presence of the signer and rejects the request early if missing.
+#[must_use]
+fn with_unsealed_guard(ctx: context::Context) -> BoxedFilter<()> {
```

Comment:
I think it would make sense to return the `signer` here. This means that you wouldn't repeat this check inside the handlers below

Reply:
It would and it was done initially. The issue that comes from that is, as we want to use it as a guard on the top level `filters`, it would require every handle down chain to expect a `signer` as the first argument. Most don't need one, tho. See https://github.com/radicle-dev/radicle-upstream/pull/1008#pullrequestreview-503901412 for more context of how to address that issue more generally.

Reply:
That was my initial thought as well, but passing it down to the filters in each module was not straightforward. It would also mean having to handle it as an input in all handlers with the guard wether they use it or not. So I went with this implementation as a first version.

Reply:
Why wouldn't it being only added to the filters that require a signer then? :eyes:

Reply:
As far as I understand we want to guard almost all requests if the signer is not provided (later on this means no key store exists or it is sealed). Listing all projects for a user, for instance, does not need a signer, but should only be possible with an unsealed key store

Reply:
Mmmmmk, well I think it might be easy enough to express `Context` in terms of sealed and unsealed. It's everybody's favourite technique -- a type parameter :tada: 

```rust
pub struct Context<S> {
  signer: S,
}

pub type Sealed = PhantomData<!>;
pub type Unsealed = BoxedSigner;

impl Context<Sealed> {
  pub fn new(...) -> Self

  pub fn unseal(...) -> Context<Unsealed>
}
```

Then `with_unsealed_guard` would convert a `Context<Sealed>` into `Context<Unsealed>`. 

However, I think we'd end up closer to @xla's enum suggestion again since the `Context` needs to be mutated so that when we unseal it then it stays unsealed across endpoints. So we'd probably actually have `with_unsealed_guard: Either<Context<Sealed>, Context<Unsealed>> -> Context<Unsealed>`.

The good thing being that in tests we can always construct `Context<Unsealed>` and don't need to unwrap.

Do we think it's worth putting in this upfront work here and now? The advantage being that we don't have to go back and start removing unwraps et al. 

P.S. This is how `Storage` works in `radicle-link` if you want to take a look there.

Reply:
I'd prefer to do it in a later PR. We won't need the `Empty` state until #999 and I think we will gain more insight into it once we start implementing #998.

Reply:
Okey dokey :+1: